debian/rules

```diff
@@ -7,110 +7,23 @@ export DH_VERBOSE=1
 
 package=logitechmediaserver
 base=$(CURDIR)/debian/$(package)
-share=usr/share/squeezeboxserver/
 source=$(CURDIR)/../server
 
-varlib=var/lib/squeezeboxserver
+# main packaging script based on dh7 syntax
+%:
+	dh $@ 
 
+# Overrides for variances within Logitech Media Server file locations
+override_dh_perl:
+	# using hard-coded dependency in control file
 
-clean:
-	dh_testdir
-	dh_testroot
-	rm -rf ${base}
-
-install:
-	dh_testdir
-	dh_testroot
-	dh_installdirs
-
-	# Create directories needed.
-	install -d -m0755 ${base}/etc/default/
-	install -d -m0755 ${base}/etc/squeezeboxserver/
-	install -d -m0755 ${base}/etc/init.d/
-	install -d -m0755 ${base}/usr/share/perl5/
-	install -d -m0755 ${base}/${share}
-	install -d -m0755 ${base}/usr/share/lintian/overrides/
-	install -d -m0755 ${base}/usr/share/doc/squeezeboxserver/
-	install -d -m0755 ${base}/${varlib}/prefs
-	install -d -m0755 ${base}/${varlib}/cache
+override_dh_installdocs:
+	cp ${source}/License.txt $(CURDIR)/debian/copyright
+	dh_installdocs
 
+override_dh_install:
+	dh_install -X darwin -X freebsd -X MSWin -X sparc-linux -X PreventStandby -X errmsg
 	# Copy our server files to the appropriate Debian locations.
 	install -m0755 ${source}/slimserver.pl ${base}/usr/sbin/squeezeboxserver
 	install -m0755 ${source}/scanner.pl ${base}/usr/sbin/squeezeboxserver-scanner
 	install -m0755 ${source}/cleanup.pl ${base}/usr/sbin/squeezeboxserver-cleanup
-
-	cp -r ${source}/Slim ${base}/usr/share/perl5/
-	
-	# Copy our CPAN directory and locally modified perl modules
-	cp -r ${source}/CPAN ${base}/${share}
-	cp -r ${source}/lib ${base}/${share}
-
-	# Make sure we copy in the Bin directory
-	cp -r ${source}/Bin ${base}/${share}
-
-	# Remove non-Linux binary modules and binaries
```

Comment:
The -X options to dh_install exclude the incompatible files.

By itself, I don't have a great reason for the Debian update.  But, it does
make splitting the package into the pure perl and arch specific bits
easier, which is what I'll work on next.  Updating the current state was a
good way to build familiarity with package building for Debian.

On Mon, Nov 9, 2015, 9:10 PM mherger <notifications@github.com> wrote:

> In debian/rules
> <https://github.com/Logitech/slimserver-platforms/pull/2#discussion_r44370788>
> :
>
> >  	# Copy our server files to the appropriate Debian locations.
> >  	install -m0755 ${source}/slimserver.pl ${base}/usr/sbin/squeezeboxserver
> >  	install -m0755 ${source}/scanner.pl ${base}/usr/sbin/squeezeboxserver-scanner
> >  	install -m0755 ${source}/cleanup.pl ${base}/usr/sbin/squeezeboxserver-cleanup
> > -
> > -	cp -r ${source}/Slim ${base}/usr/share/perl5/
> > -	
> > -	# Copy our CPAN directory and locally modified perl modules
> > -	cp -r ${source}/CPAN ${base}/${share}
> > -	cp -r ${source}/lib ${base}/${share}
> > -
> > -	# Make sure we copy in the Bin directory
> > -	cp -r ${source}/Bin ${base}/${share}
> > -
> > -	# Remove non-Linux binary modules and binaries
>
> I don't see where in the new build configuration this would be handled?
> Are incompatible files no longer removed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Logitech/slimserver-platforms/pull/2/files#r44370788>.
>